src/compiler-libs-406/ident.ml

```diff
@@ -55,7 +55,7 @@ let same i1 i2 = i1 = i2
        then i1.stamp = i2.stamp
        else i2.stamp = 0 && i1.name = i2.name *)
 
-let compare i1 i2 = Pervasives.compare i1 i2
+let compare i1 i2 = Stdlib.compare i1 i2
```

Comment:
I hit the same issue.
My workaround is just to config the build ignore deprecation warnings